File: pdam-backend/pdam.api/src/main/java/com/salesianostriana/dam/pdam/api/festival/dto/GetFestivalDto.java
package com.salesianostriana.dam.pdam.api.festival.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.salesianostriana.dam.pdam.api.festival.model.Festival;
import com.stripe.model.PaymentIntent;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class GetFestivalDto {

    private Long id;
    private String name;
    private String description;
    private String location;

    @JsonFormat(pattern = "dd/MM/yyyy HH:mm:ss")
    private LocalDateTime date;

    private int duration;
    private int remainingTickets;
    private String city;
    private double price;
    private boolean drinkIncluded;
    private int numberOfDrinks;
    private boolean adult;
    private String imgPath;
    private String stripeId;

    public static GetFestivalDto of(Festival festival) {

        return GetFestivalDto.builder()
                .id(festival.getId())
                .description(festival.getDescription())
                .name(festival.getName())
                .location(festival.getLocation())
                .date(festival.getDateTime())
                .duration(festival.getDuration())
                .remainingTickets(festival.getCapacity()-festival.getClients().size())
                .city(festival.getCity().getName())
                .price(festival.getPrice())
                .drinkIncluded(festival.isDrinkIncluded())
                .numberOfDrinks(festival.getNumberOfDrinks())
                .adult(festival.isAdult())
                .imgPath(festival.getImgPath())
                .build();
    }

    public static GetFestivalDto ofStripe(Festival festival, String stripeId) {
        return GetFestivalDto.builder()
                .id(festival.getId())
                .description(festival.getDescription())
                .name(festival.getName())
                .location(festival.getLocation())
                .date(festival.getDateTime())
                .duration(festival.getDuration())
                .remainingTickets(festival.getCapacity()-festival.getClients().size())
                .city(festival.getCity().getName())
                .price(festival.getPrice())
                .drinkIncluded(festival.isDrinkIncluded())
                .numberOfDrinks(festival.getNumberOfDrinks())
                .adult(festival.isAdult())
                .stripeId(stripeId)
                .build();
    }
}

File: pdam-frontend/pdam_app/lib/models/event/GetFestivalDto.dart
class GetFestivalDto {
  int? id;
  String? name;
  String? description;
  String? location;
  String? date;
  int? duration;
  int? remainingTickets;
  String? city;
  double? price;
  bool? drinkIncluded;
  int? numberOfDrinks;
  bool? adult;
  String? imgPath;
  String? stripeId;

  GetFestivalDto(
      {this.id,
      this.name,
      this.description,
      this.location,
      this.date,
      this.duration,
      this.remainingTickets,
      this.city,
      this.price,
      this.drinkIncluded,
      this.numberOfDrinks,
      this.adult,
      this.imgPath,
      this.stripeId});

  GetFestivalDto.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    description = json['description'];
    location = json['location'];
    date = json['date'];
    duration = json['duration'];
    remainingTickets = json['remainingTickets'];
    city = json['city'];
    price = json['price'];
    drinkIncluded = json['drinkIncluded'];
    numberOfDrinks = json['numberOfDrinks'];
    adult = json['adult'];
    imgPath = json['imgPath'];
    stripeId = json['stripeId'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['name'] = this.name;
    data['description'] = this.description;
    data['location'] = this.location;
    data['date'] = this.date;
    data['duration'] = this.duration;
    data['remainingTickets'] = this.remainingTickets;
    data['city'] = this.city;
    data['price'] = this.price;
    data['drinkIncluded'] = this.drinkIncluded;
    data['numberOfDrinks'] = this.numberOfDrinks;
    data['adult'] = this.adult;
    data['imgPath'] = this.imgPath;
    data['stripeId'] = this.stripeId;
    return data;
  }
}
